app/controllers/api/v2/accredited_body_training_provider_courses_controller.rb

```diff
@@ -0,0 +1,37 @@
+module API
+  module V2
+    class AccreditedBodyTrainingProviderCoursesController < ApplicationController
+      before_action :build_recruitment_cycle, :build_accredited_body, :build_training_provider
+
+      def index
+        authorize accredited_body, :can_list_training_providers?
+
+        course_scope = accredited_body.current_training_provider_courses(training_provider.id)
+        courses = CoursePolicy::TrainingProviderCourses.new(current_user, course_scope).resolve
+        render jsonapi: courses, include: params[:include]
+      end
+
+    private
+
+      attr_reader :recruitment_cycle, :accredited_body, :training_provider
+
+      def build_recruitment_cycle
+        @recruitment_cycle = RecruitmentCycle.find_by(
+          year: params[:recruitment_cycle_year],
+        ) || RecruitmentCycle.current_recruitment_cycle
+      end
+
+      def build_accredited_body
+        @accredited_body = recruitment_cycle.providers.find_by!(
```

Comment:
```suggestion
        @provider = recruitment_cycle.providers.find_by!(
```

Reply:
Similar to a suggestion below, I'm voting for `accredited_body` because I think it adds useful context.